test_scripts/Smoke/API/025_UnsubscribeVehicleData_PositiveCase.lua

```diff
@@ -0,0 +1,152 @@
+-- c. appID is currently in Background, Full or Limited HMI level
+
+-- Steps:
+-- appID requests UnsubscribeVehicleData with valid parameters
+
+-- Expected:
+-- SDL validates parameters of the request
+-- SDL checks if VehicleInfo interface is available on HMI
+-- SDL checks if UnsubscribeVehicleData is allowed by Policies
+-- SDL checks if all parameters are allowed by Policies
+-- SDL transfers the VehicleInfo part of request with allowed parameters to HMI
+-- SDL receives VehicleInfo part of response from HMI with "SUCCESS" result code
+-- SDL responds with (resultCode: SUCCESS, success:true) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSmoke = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Local Variables ]]
+local VDValues = {
+  gps = "VEHICLEDATA_GPS",
+  speed = "VEHICLEDATA_SPEED",
+  rpm = "VEHICLEDATA_RPM",
+  fuelLevel = "VEHICLEDATA_FUELLEVEL",
+  fuelLevel_State = "VEHICLEDATA_FUELLEVEL_STATE",
+  instantFuelConsumption = "VEHICLEDATA_FUELCONSUMPTION",
+  externalTemperature = "VEHICLEDATA_EXTERNTEMP",
+  prndl = "VEHICLEDATA_PRNDL",
+  tirePressure = "VEHICLEDATA_TIREPRESSURE",
+  odometer = "VEHICLEDATA_ODOMETER",
+  beltStatus = "VEHICLEDATA_BELTSTATUS",
+  bodyInformation = "VEHICLEDATA_BODYINFO",
+  deviceStatus = "VEHICLEDATA_DEVICESTATUS",
+  driverBraking = "VEHICLEDATA_BRAKING",
+  wiperStatus = "VEHICLEDATA_WIPERSTATUS",
+  headLampStatus = "VEHICLEDATA_HEADLAMPSTATUS",
+  engineTorque = "VEHICLEDATA_ENGINETORQUE",
+  accPedalPosition = "VEHICLEDATA_ACCPEDAL",
+  steeringWheelAngle = "VEHICLEDATA_STEERINGWHEEL",
+  eCallInfo = "VEHICLEDATA_ECALLINFO",
+  airbagStatus = "VEHICLEDATA_AIRBAGSTATUS",
+  emergencyEvent = "VEHICLEDATA_EMERGENCYEVENT",
+  clusterModeStatus = "VEHICLEDATA_CLUSTERMODESTATUS",
+  myKey = "VEHICLEDATA_MYKEY"
+}
+
+local paramsList = {}
+local requestParams = { }
+for k, _ in pairs(VDValues) do
+  table.insert(paramsList, k)
+  requestParams[k] = true
+end
+
+local responseUiParams = { }
+
+local allParams = {
+  requestParams = requestParams,
+  responseUiParams = responseUiParams,
+}
+
+--[[ Local Functions ]]
+local function ptuUpdateFunc(tbl)
+  local SVDgroup = {
+    rpcs = {
+      SubscribeVehicleData = {
+        hmi_levels = { "BACKGROUND", "FULL", "LIMITED" },
+        parameters = paramsList
+      },
+      UnsubscribeVehicleData = {
+        hmi_levels = { "BACKGROUND", "FULL", "LIMITED" },
+        parameters = paramsList
+      }
+    }
+  }
+  tbl.policy_table.functional_groupings.NewTestCaseGroup = SVDgroup
+  tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.appID].groups =
+  { "Base-4", "NewTestCaseGroup" }
+end
+
+local function setVDResponse()
+  local temp = { }
+  local vehicleDataResultCodeValue = "SUCCESS"
+  for k, v in pairs(VDValues) do
```

Comment:
@GetmanetsIrina Please avoid to use obfuscated names for variables which returns by iterators in case them is used in logic.
In other words: what are `k` and `v` here?

Reply:
@aderiabin , Find update related to all comments in d5d3d3d0a890b0291e8dddbf8266d6150c081b01.